src/eventDispatchers/mouseEventHandlers/mouseDown.js

```diff
@@ -29,31 +33,133 @@ export default function (evt) {
 
   let tools;
   const eventData = evt.detail;
-  const element = eventData.element;
-  const coords = eventData.currentPoints.canvas;
-  const distance = 6;
+  const element = evt.detail.element;
+  const coords = evt.detail.currentPoints.canvas;
 
-  // TODO: instead of filtering these for every interaction, we can change our
-  // State's structure to always know these values.
-  // Filter out disabled and enabled
+  // High level filtering
   tools = getInteractiveToolsForElement(element, getters.mouseTools());
-  // Filter out tools that do not match mouseButtonMask
   tools = tools.filter((tool) =>
     isMouseButtonEnabled(eventData.which, tool.options.mouseButtonMask)
   );
-  tools = getToolsWithDataForElement(element, tools);
 
-  // Find tools with handles we can move
-  const toolsWithMoveableHandles = tools.filter((tool) => {
-    const toolState = getToolState(eventData.element, tool.name);
+  const activeTools = tools.filter((tool) => tool.mode === 'active');
+
+  // If any tools are active, check if they have a special reason for dealing with the event.
+  if (activeTools.length > 0) {
+    // TODO: If length > 1, you could assess fitness and select the ideal tool
+    // TODO: But because we're locking this to 'active' tools, that should rarely be an issue
+    const firstActiveTool = activeTools[0];
+
+    if (typeof firstActiveTool.activeMouseDownCallback === 'function') {
+      const shouldPreventPropagation = firstActiveTool.activeMouseDownCallback(evt);
```

Comment:
is `activeMouseDownCallback` to indicate it's only called for `active` tools?
That makes sense, but I initially wondered if it was for the `mouseDownActivate` event

Reply:
Hey, you're correct that that's what I was trying to dictate, but I didn't think of that, maybe the naming is a little confusing.